src/esw/brushed_motors/Controller.cpp

```diff
@@ -197,12 +198,34 @@ void Controller::turnOn() const {
 }
 
 // REQUIRES: nothing
-// MODIFIES: isLive
+// MODIFIES: liveMap
 // EFFECTS: I2C bus, if not already live,
 // configures the physical controller.
 // Then makes live.
+
+std::unordered_map<uint8_t, LiveState> Controller::liveMap;
+
 void Controller::makeLive() {
-    if (isLive) {
+    // if (isLive) {
+    //     return;
+    // }
+
+    uint8_t key = (deviceAddress << 3) | motorID;
+
+    // if key is absent, no motor with that key has been made live
+    auto it = liveMap.find(key);
+    if (it != liveMap.end()) {
+        it->second.liveMutex.lock();
+    } else {
+        LiveState state;
+        state.jointName = name;
+        liveMap[key] = state;
```

Comment:
It might be better to emplace a LiveState object into the map, right now this is copying which I think should work find, but is a little unnecessary.